etl/jobs/run_change_detection_batches.py

```diff
@@ -0,0 +1,158 @@
+import sys
+import math
+import boto3
+import time
+from awsglue.transforms import *
+from awsglue.utils import getResolvedOptions
+from pyspark.context import SparkContext
+from awsglue.context import GlueContext
+from awsglue.job import Job
+
+contexts = {
+    "england": {
+        "workflow_name": "generate-compare-nearest-50",
+        "aggregation_trigger_name": "Run aggregate-monthly",
+        "aggregation_job_name": "aggregate-monthly-parameterised"
+    },
+    "england_test": {
+        "workflow_name": "generate-compare-nearest-50-test",
+        "aggregation_trigger_name": "Run Aggregate Monthly Test",
+        "aggregation_job_name": "aggregate-monthly-parameterised"
+    },
+    "scotland": {
+        "workflow_name": "generate-compare-nearest-50-scotland",
+        "aggregation_trigger_name": "Run Aggregate Monthly Scotland",
+        "aggregation_job_name": "aggregate-monthly-parameterised"
+    },
+    "scotland_test": {
+        "workflow_name": "generate-compare-nearest-50-scotland-test",
+        "aggregation_trigger_name": "Run Aggregate Monthly Test for Scotland",
+        "aggregation_job_name": "aggregate-monthly-parameterised"
+    },
+}
+
+def get_date_ranges(from_year, from_month, to_year, to_month, months_per_run):
+    date_ranges = []
+    total_months = (to_year - from_year) * 12 + to_month - from_month + 1
+    no_runs = math.ceil(total_months / months_per_run)
+    if (no_runs < 1): exit("From and To date ranges not specified correctly") 
+    start_year = from_year
+    start_month = from_month
+    for i in range(no_runs):
+       end_year_month = add_months_with_floor(start_year, start_month, months_per_run - 1, to_year, to_month)
+       date_range = {
+          "--FROM_YEAR_MONTH": format_year_month_as_parameter(start_year, start_month),
+          "--TO_YEAR_MONTH": format_year_month_as_parameter(end_year_month[0], end_year_month[1])
+       }
```

Comment:
My brain is not taking in all this maths..! Have you tried running it against weird cases like 2020-12 2021-01?

Reply:
Yep, I tested it against all the edge cases I could think of, for example going over year boundaries as you say.  The one thing I didn't think of was if they specify month not between 1 and 12, that would definitely make the maths do weird things here, but I will catch that in the initial validation as you specified in the other comment